tests/test_data_validation_before_insert.py

```diff
@@ -0,0 +1,35 @@
+import os
+import sys
+import pandas as pd
+import logging
+from shared_code.utils import validate_data
+
+logger = logging.getLogger(__name__)
+
+_i = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+if _i not in sys.path:
+    # add parent directory to sys.path so utils module is accessible
+    sys.path.insert(0, _i)
+del _i  # clean up global name space
+
+
+# This scrapper tests if the script validator can remove any rows with invalid data
+class TestDataScraper():
+    def test_validator_on_rows(self):
+        df = pd.read_csv('tests/test_data_validation_before_insert.csv')
+        validate_data(df)
+        logger.info(df)
+        logger.info(" The dataframe contains " + str(len(df)) + " valid rows. 13 valid rows were expected.")
```

Comment:
Having the logger is not sufficient for the test. If the test fails you want pytest module to abandon the build because the test failed. With just the logger the test fails silently and build script wouldn't notice it either.

You need assert statement like this     `assert len(df) == 13, 'Data frame should have 13 valid rows but found ' + len(df)`
So if the logic in validate_data() is changed and if it produces lesser than 13 or higher than 13 rows the assert statement will fail causing the test to fail and hence the build script.